compiler/src/dotty/tools/dotc/typer/PrepareTransparent.scala

```diff
@@ -0,0 +1,411 @@
+package dotty.tools
+package dotc
+package typer
+
+import dotty.tools.dotc.ast.Trees.NamedArg
+import dotty.tools.dotc.ast.{Trees, untpd, tpd, TreeTypeMap}
+import Trees._
+import core._
+import Flags._
+import Symbols._
+import Types._
+import Decorators._
+import Constants._
+import StdNames.nme
+import Contexts.Context
+import Names.{Name, TermName, EmptyTermName}
+import NameOps._
+import NameKinds.{ClassifiedNameKind, InlineAccessorName, UniqueInlineName}
+import ProtoTypes.selectionProto
+import SymDenotations.SymDenotation
+import Annotations._
+import transform.{ExplicitOuter, AccessProxies}
+import Inferencing.fullyDefinedType
+import config.Printers.inlining
+import ErrorReporting.errorTree
+import collection.mutable
+import transform.TypeUtils._
+import reporting.trace
+import util.Positions.Position
+import util.Property
+import ast.TreeInfo
+
+object PrepareTransparent {
+  import tpd._
+
+  /** Marks an implicit reference found in the context (as opposed to the implicit scope)
+   *  from an inlineable body. Such references will be carried along with the body to
+   *  the expansion site.
+   */
+  private val ContextualImplicit = new Property.StickyKey[Unit]
+
+  /** An attachment labeling a toplevel match node of a transparent function */
```

Comment:
It'd be nice to define what we mean by "toplevel" here.

Reply:
Toplevel is defined in `typelevel.md`, which will become part of the language spec. I believe we can therefore use it in the code without further explanation.